std/random.d

```diff
@@ -1183,6 +1183,130 @@ A singleton instance of the default random number generator
 }
 
 /**
+A random bool generator that has the underlying random integral
+generation batched.
+ */
+struct RandomBool(Rng = Random)
+    if (isUniformRNG!Rng || isUniformRNG!(typeof(*Rng.init)))
+{
+    private Rng parent;
+    private uint mask;
+    private uint accumulator;
+
+    /// Always `false`
+    enum bool empty = false;
+
+    static if (isPointer!Rng)
+    {
+        this(Rng rng) @safe
+        {
+            parent = rng;
+            initPrivates();
+        }
+    }
+    else
+    {
+        /**
+          If Rng is not a pointer, we want to pass it by ref
+          in order to avoid two passByValues
+          */
+        this(ref Rng rng) @safe
+        {
+            parent = rng;
+            initPrivates();
+        }
+    }
```

Comment:
I might have been wrong about advising you to use static if. How about a single constructor:

```d
this()(auto ref Rng rng)
{
    ...
}
```

This constructor is (subtly) a parameterless template, which allows you to use `auto ref`.